### ISSUE TITLE
Example for basic multiple SQL database configure JSON and query? 基本多 SQL 数据库配置 JSON 和查询的示例？

### ISSUE THREAD

**Issue by GavinRay97** [MyCATApache/Mycat2#667]
Hello, apologies if the translation for this is poor, I am using Google Translate.
你好，如果翻译不好，我很抱歉，我正在使用谷歌翻译。

I am curious if it is possible to do something like the below?
我很好奇是否可以执行以下操作？

```mermaid
flowchart LR
    MySQL_Client
    subgraph App
        direction LR
        oracle[(Oracle)]
        mongo[(Mongo)]
        mysql[(MySQL)]
        MyCAT2
        MyCAT2<-->oracle
        MyCAT2<-->mongo
        MyCAT2<-->mysql
    end
    MySQL_Client-. "SELECT ... FROM mysql.db1 \n INNER JOIN oracle.db2 \n INNER JOIN mongo.db3 " .->App
```

From reading the docs/source, it seems yes:
通过阅读文档/源代码，似乎是的

- https://www.yuque.com/ccazhw/ml3nkf/xzh6n5

https://github.com/MyCATApache/Mycat2/blob/c6aacf671c073e324c97fcdf6e6ca9632b97cf40/config/src/main/java/io/mycat/config/DatasourceConfig.java#L45-L58

But I am not able to understand the documentation enough from poor translation to see how to configure MyCAT2 to do this.
但是我无法从糟糕的翻译中充分理解文档，以了解如何配置 MyCAT2 来执行此操作。

Thank you =)
谢谢你 =)

**Comment by zwyqz** [comment]
我们的架构是支持多数据源的，但是没有进一步的适配；目前对mysql支持比较好，如果您需要配置可以参考https://www.yuque.com/ccazhw/ml3nkf/fb2285b811138a442eb850f0127d7ea3 或者https://blog.csdn.net/w329636271/article/details/115632101， 然后使用注解（https://www.yuque.com/ccazhw/ml3nkf/f9f24306bbd3992c1baff00cdb0956a4）去尝试使用mycat2；

**Comment by junwen12221** [comment]
Yes, your are right. The design of mycat2 takes MySQL protocol as the front end and various databases as the backend, it supports MySQL as the backend (mainly), and also realize a simple SQL conversion to mongodb driver, which makes mongodb driver disguise as MySQL data source. For other databases, the idea is the same, but there is no complete implementation. Only MySQL and a few simple SQL conversion to mongodb (io.mycat.datasource.jdbc.mongodb.MongoConnection) can work temporarily,  jooq can realize the conversion of data source dialect (branch main, io/mycat/datasource/jdbc/DruidDatasourceProvider.java:114), Apache Calcite can also convert (io.mycat.calcite.MycatCalciteSupport#convertToSqlTemplate), Alibaba Druid can also convert(com.alibaba.druid.sql.SQLUtils#toPGString,io/mycat/calcite/MycatCalciteSupport.java:654)


SELECT ... FROM mysql.db1
INNER JOIN oracle.db2
INNER JOIN mongo.db3

But what you said is not feasible for the time being.Because the MySQL driver disguised by Mongo may not support some SQL query syntax.


If you want to discuss further, you can email me: 294712221@qq.com Or you can talk here.
